### PR TITLE
fix(frontend): The "Back" button on the "Friend profile" page doesn't navigate to the previous tab on the "Friends" page tm-325

### PR DESCRIPTION
**Pull request by GEOFARL**

### CODE REVIEW

File: apps/frontend/src/pages/friends/libs/components/friend/friend.tsx [hunk 1]
```diff
@@ -30,9 +31,22 @@ const Friend: React.FC<Properties> = ({ friend }: Properties) => {
...
+	} as const;
+
 	const chatRouteByUser =
-		`${AppRoute.CHATS}?${parameters.toString()}` as typeof AppRoute.CHATS;
+		`${AppRoute.CHATS}?${Parameters.CHAT_PARAMETERS.toString()}` as typeof AppRoute.CHATS;
+
+	const userRouteWithRedirect =
+		`/users/${friend.id}?${Parameters.REDIRECT_TO_PARAMETERS.toString()}` as ValueOf<
+			typeof AppRoute
+		>;
```
Comment (fshabanov) [inline apps/frontend/src/pages/friends/libs/components/friend/friend.tsx:49]: I don't like that in URL we will always have `?redirectTo=...`, is there another way to do this?

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/friends/libs/components/friend/friend.tsx [hunk 2]
```diff
@@ -30,9 +31,22 @@ const Friend: React.FC<Properties> = ({ friend }: Properties) => {
 	);
 	const dispatch = useAppDispatch();
 
-	const parameters = new URLSearchParams({ user: String(friend.id) });
+	const { pathname } = useLocation();
+
+	const Parameters = {
```
Comment (what1s1ove) [inline apps/frontend/src/pages/friends/libs/components/friend/friend.tsx:36]: B35

**Review by what1s1ove — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Review by Sytnikovakatya — APPROVED** [review]

File: apps/frontend/src/pages/user/libs/constants/previous-page.constant.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+const PREVIOUS_PAGE = -1;
```
Comment (what1s1ove) [inline apps/frontend/src/pages/user/libs/constants/previous-page.constant.ts:1]: we have a constant for it `BACK_NAVIGATION_STEP`

**Review by what1s1ove — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/user/user.tsx [hunk 4]
```diff
@@ -21,6 +22,7 @@ import {
 	actions as usersActions,
 } from "~/modules/users/users.js";
 
+import { BACK_NAVIGATION_STEP } from "../course-description/libs/constants/constants.js";
```
Comment (fshabanov) [inline apps/frontend/src/pages/user/user.tsx:25]: Move this constant to libs/constants

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Review by what1s1ove — APPROVED** [review]

**Pull request merged**